### PR TITLE
Add patch that extends model metadata list to make use of custom tags

### PR DESCRIPTION
**Pull request by RafalSkolasinski**
Also adds machinery for applying the patch.

image

### CODE REVIEW

**Comment by RafalSkolasinski** [comment]
updated notebook: https://nbviewer.jupyter.org/github/RafalSkolasinski/seldon-deploy-sdk/blob/sdk-hack-tags/examples/metadata_demo.ipynb

File: examples/metadata_demo.ipynb [hunk 1]
```diff
@@ -139,95 +152,246 @@
...
+    "## List all models in the Model Catalogue"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 8,
    "metadata": {},
-   "outputs": [],
+   "outputs": [
```
Comment (ivan-valkov) [inline examples/metadata_demo.ipynb:171]: do we want the outputs to be part of the notebook? Maybe we do I am not sure, before commiting I was cleaning up the outputs with `jupyter nbconvert --ClearOutputPreprocessor.enabled=True --inplace *.ipynb`
I guess they can be helpful to see what an output looks like

**Review by ivan-valkov — APPROVED** [review]
nice one :+1:

File: examples/metadata_demo.ipynb (hunk 1 above)
Comment (RafalSkolasinski) [reply examples/metadata_demo.ipynb:171]: It's nice in the sense that when you view it on github you see more or less how it should work. 
I usually before commiting restart the kernel and launch it top to bottom.

**Pull request merged**